db/music_pg.sql

```diff
@@ -0,0 +1,373 @@
+  "artist" text NOT NULL,
+  "collaborators" text,
+  "composer" text,
+  "year_min" int,
+  "year_max" int
+);
+
+CREATE TABLE "song_relation" (
+  "song_relation_id" INT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+  "song_from_id" int NOT NULL,
+  "song_to_id" int NOT NULL,
+  "relation_type" text NOT NULL
+);
+
+CREATE TABLE "song_artist" (
+  "song_artist_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "song_id" int NOT NULL,
+  "artist_id" int NOT NULL,
+  "credit_type" text NOT NULL
+);
+
+CREATE TABLE "artist" (
+  "artist_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "name" text NOT NULL,
+  "year_min" int,
+  "year_max" int,
+  "primary_artist_id" int
+);
+
+CREATE TABLE "artist_genre" (
+  "artist_genre_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "artist_id" int NOT NULL,
+  "genre_id" int NOT NULL
+);
+
+CREATE TABLE "participant" (
+  "participant_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "username" text UNIQUE NOT NULL,
+  "nickname" text,
+  "date_joined" timestamp NOT NULL DEFAULT (now()),
+  "time_zone" text,
+  "active" bool_int NOT NULL
+);
+
+CREATE TABLE "participant_nickname" (
+  "participant_nickname_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "participant_id" int NOT NULL,
+  "nickname" text NOT NULL,
+  "date_start" date NOT NULL,
+  "date_end" date
+);
+
+CREATE TABLE "genre" (
+  "genre_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "name" text UNIQUE NOT NULL,
+  "parent_id" text
+);
+
+CREATE TABLE "song_genre" (
+  "song_genre_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "song_id" int NOT NULL,
+  "genre_id" int NOT NULL
+);
+
+CREATE TABLE "bracket" (
+  "bracket_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "epoch" int,
+  "cycle" int,
+  "name" text,
+  "type" text NOT NULL,
+  "year_min" int,
+  "year_max" int,
+  "order" int NOT NULL,
+  "size" text NOT NULL,
+  "date_start" timestamp,
+  "date_end" timestamp,
+  "discord_message_id" text
+);
+
+CREATE TABLE "project" (
+  "project_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "name" text UNIQUE,
+  "project_lead_id" int
+);
+
+CREATE TABLE "bracket_selection" (
+  "bracket_selection_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "project_id" int NOT NULL,
+  "epoch" int NOT NULL,
+  "cycle" int NOT NULL,
+  "bracket_id" int NOT NULL,
+  "date_selected" timestamp NOT NULL
+);
+
+CREATE TABLE "seed" (
+  "seed_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "bracket_id" int NOT NULL,
+  "seed" int NOT NULL,
+  "submission_id" int NOT NULL,
+  "emoji" text
+);
+
+CREATE TABLE "seed_history" (
+  "seed_history_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "date_changed" timestamp NOT NULL,
+  "reason_changed" text NOT NULL,
+  "seed_id" int NOT NULL,
+  "submission_id" int NOT NULL,
+  "emoji" text
+);
+
+CREATE TABLE "round" (
+  "round_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "bracket_id" int NOT NULL,
+  "name" text NOT NULL,
+  "size" int NOT NULL,
```

Comment:
We may want to allow this to be null, e.g. in the playin round

Reply:
Ah I see the other note I had forgotten about that, I'm fine leaving it as non-null and just using 0 as the placeholder value for it.